### PR TITLE
Fix a false positive for ``no-member``.

### LINKED ISSUE #8759: Missing Member for Pydantic Model
### Bug description

hey team, I'm running into:
```E1101: Instance of 'Response' has no 'versionInfo' member (no-member)```
Provided the following code:
```
☁  test_py  cat main.py 
from response import Response
from run import f

r = f()
r.versionInfo = r.versionInfo | {"hi":"bye"}
print(r)%                                                                                                                                                                                     ☁  test_py  cat response.py 
from pydantic import BaseModel


class Response(BaseModel):
    versionInfo: dict[str, str]
    fm: dict[int, float]
☁  test_py  cat run.py 
from response import Response

def f() -> Response:
  return Response(verionInfo={}, fm={})%
```
Running pylint on `main.py` I get the following:
```main.py:5:16: E1101: Instance of 'Response' has no 'versionInfo' member (no-member)```

However, `versionInfo` is defined in the `Response` class.

I've got a theory for what's wrong and a work around. Specifically, I think it has something to do with how the `BaseModel` in pydantic isn't defining the fields using self / something more python native. When I replace the `BaseModel` with a normal class and define the fields with `self` linting passes.

Curious what the team thought should be done here. It's not strictly a bug given it's related to a separate dependency in pydantic.

### Configuration

```ini
pylint 2.17.0
astroid 2.15.0
Python 3.9.6 (default, May  7 2023, 23:32:44) 
[Clang 14.0.3 (clang-1403.0.22.14.1)]
```

### Command used

```shell
☁  test_py  pylint main.py
```

### Pylint output

```shell
************* Module main
main.py:6:0: C0304: Final newline missing (missing-final-newline)
main.py:1:0: C0114: Missing module docstring (missing-module-docstring)
main.py:5:16: E1101: Instance of 'Response' has no 'versionInfo' member (no-member)
main.py:1:0: W0611: Unused Response imported from response (unused-import)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

************* Module main
main.py:6:0: C0304: Final newline missing (missing-final-newline)
main.py:1:0: C0114: Missing module docstring (missing-module-docstring)
main.py:1:0: W0611: Unused Response imported from response (unused-import)

^ no more:
main.py:5:16: E1101: Instance of 'Response' has no 'versionInfo' member (no-member)
or the option to do away with it.

### Pylint version

```shell
pylint 2.17.0
astroid 2.15.0
Python 3.9.6 (default, May  7 2023, 23:32:44) 
[Clang 14.0.3 (clang-1403.0.22.14.1)]
```

### OS / Environment

ProductName:		macOS
ProductVersion:		13.4
BuildVersion:		22F66

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by mbyrnepr2**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
| ✓   | :sparkles: New feature |
|    | :hammer: Refactoring   |
|    | :scroll: Docs          |

## Description

Fix a false positive for ``no-member`` when accessing a no-value attribute that is annotated in a class body.

Closes #8759

### CODE REVIEW

**Comment by Pierre-Sassoulas** [comment]
Is it still a draft @mbyrnepr2 ?

**Comment by mbyrnepr2** [comment]
Thanks for the reminder @Pierre-Sassoulas. It’s been a while since I thought about it and I think I must have had a doubt about it to leave it as a draft. 
A review/feedback would be helpful to see if this change makes sense.

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
I think we need to do something specific to pydantic(s Base Model to treat it correctly erven if it's dynamic (maybe a brain in astroid, maybe an external plugin) but I'm not a pydantic user so I don't know exactly how BaseModel is supposed to behave.

File: tests/functional/n/no/no_member_augassign.py [hunk 1]
```diff
@@ -1,25 +1,30 @@
...
+
 
 # Test for: https://github.com/pylint-dev/pylint/issues/4562
 class A:
     value: int
 
+
 obj_a = A()
-obj_a.value += 1  # [no-member]
+obj_a.value += 1
```
Comment (Pierre-Sassoulas) [inline tests/functional/n/no/no_member_augassign.py:13]: A review comment got lost: Here we'd introduce a false negative, running the code raise an error:
```
Traceback (most recent call last):
  File "/home/pierre/pylint/doc/a.py", line 12, in <module>
    obj_a = A(1)
TypeError: A() takes no arguments
```

**Pull request closed without merging**